File: frontend/src/types/organization.ts
export interface Organization {
  id: string;
  name: string;
  slug: string;
  is_personal: boolean;
  created_by: string;
  created_at: string;
  updated_at: string;
}

export interface ProjectMetrics {
  document_count: number;
  gap_count: number;
  pending_gap_count: number;
  avg_rating: number | null;
  healthy_percentage: number;
}

export interface Project {
  id: string;
  organization_id: string;
  name: string;
  slug: string;
  description: string | null;
  created_by: string;
  created_at: string;
  updated_at: string;
  metrics?: ProjectMetrics;
}

export interface CreateOrganizationDto {
  name: string;
  slug: string;
}

export interface CreateProjectDto {
  organization_id: string;
  name: string;
  slug: string;
  description?: string;
}

File: backend/shared/schemas/organization.py
"""Organization schemas for API request/response validation."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class OrganizationBase(BaseModel):
    """Base organization schema with common fields."""

    name: str = Field(
        ..., min_length=1, max_length=255, description="Organization name"
    )
    slug: str = Field(..., min_length=1, max_length=50, description="URL-safe slug")


class OrganizationCreate(OrganizationBase):
    """Schema for creating an organization."""

    is_personal: bool = Field(
        default=False, description="Whether this is a personal organization"
    )


class OrganizationResponse(BaseModel):
    """Schema for organization response."""

    id: UUID = Field(description="Organization ID")
    name: str = Field(description="Organization name")
    slug: str = Field(description="URL-safe slug")
    is_personal: bool = Field(description="Whether this is a personal organization")
    created_by: UUID = Field(description="Creator user ID")
    created_at: datetime = Field(description="Creation timestamp")
    updated_at: datetime = Field(description="Last update timestamp")

    model_config = {"from_attributes": True}


class OrganizationListItem(BaseModel):
    """Schema for organization list item."""

    id: UUID = Field(description="Organization ID")
    name: str = Field(description="Organization name")
    slug: str = Field(description="URL-safe slug")
    is_personal: bool = Field(description="Whether this is a personal organization")
    created_at: datetime = Field(description="Creation timestamp")

    model_config = {"from_attributes": True}
